parsec/backend/asgi/rpc.py

```diff
@@ -268,3 +281,138 @@ async def authenticated_api(raw_organization_id: str) -> Response:
 
     cmd_rep = await cmd_func(client_ctx, msg)
     return _rpc_msgpack_rep(cmd_rep, api_version)
+
+
+# SSE in Quart-Trio is a bit more complicated than expected:
+# In theory we would just have to return as body an async iterator that yields
+# bytes each time it needs to provide a SSE event.
+# However in practice this doesn't work at all given async iterator doesn't
+# properly support trio scopes (e.g. using a cancel scope in an async iterator
+# won't work as expected !).
+# The proper way to combine scope management and async iterator is to first have
+# an async context manager that itself provides an async iterator, the latter being
+# just a dummy channel (so that no scope logic is done in the async iterator).
+# Fortunatly Quart-Trio knows about this shortcoming, and wraps the user-provided
+# async iterator into an `IterableBody` that is used as async context manager
+# returning the async iterator.
+# So long story short, given we need to have access on the async context manager
+# (so that we close the connection's scope in case backpressure gets too high),
+# we have to implement our own custom `IterableBody` here.
+
+
+class SSEResponseIterableBody(ResponseBody):
+    def __init__(self, client_ctx: AuthenticatedClientContext, backend: BackendApp) -> None:
+        # Cancel scope will be closed automatically in case of backpressure issue
+        # Note we create it here instead of in `_make_contextmanager` to avoid concurrency
+        # issue if the cancellation kicks in between the moment we return the
+        # response object and when quart starts entering the async context manager
+        # returned by `_make_contextmanager`.
+        assert client_ctx.cancel_scope is None
+        client_ctx.cancel_scope = trio.CancelScope()
+        # Zero-sized channel to avoid backpressure issue: send is blocking until a receive occurs
+        self._body_line_sender, self._body_line_receiver = trio.open_memory_channel[bytes](0)
+        self._contextmanager = self._make_contextmanager(client_ctx, backend)
+
+    @asynccontextmanager
+    async def _make_contextmanager(
+        self, client_ctx: AuthenticatedClientContext, backend: BackendApp
+    ) -> AsyncIterator[None]:
+        async def _events_into_body_lines() -> None:
```

Comment:
I'm not sure I'd call them `lines` as a single SSE can be multiline, e.g:
```
data: This is a message, it
data: has two lines.
```
or:
```
event: remove
data: 2153
```

Reply:
I've renamed `body_line` into `sse_payload` (I'd rather not call that `event` given we already have the Parsec event that are also used in the code)